app/routes/users/charity_position.js

```diff
@@ -0,0 +1,98 @@
+import AuthorizeRoute from "../authorize";
+import Ember from "ember";
+
+export default AuthorizeRoute.extend({
+  queryParams: {
+    id: {
+      refreshModel: true
+    },
+    organisationId: {
+      refreshModel: true
+    }
+  },
+  organisationsUserService: Ember.inject.service(),
+
+  async model(params) {
+    if (params.id) {
+      return (
+        this.store.peekRecord("organisations_user", params.id) ||
+        (await this.store.findRecord("organisations_user", params.id, {
+          reload: true
+        }))
+      );
+    } else {
+      return this.store.createRecord("organisations_user", {
+        position: "",
+        preferredContactNumber: "",
+        organisationId: "",
+        user_id: "",
+        status: ""
+      });
+    }
+  },
+
+  async beforeModel(transition) {
+    const userId = transition.params["users.charity_position"].user_id;
+    this.set("user_id", userId);
+    this.store.peekRecord("user", userId) ||
+      (await this.store.findRecord("user", userId, {
+        reload: true
+      }));
+  },
+
+  async afterModel(_model, transition) {
+    const organisationId = transition.queryParams["organisationId"];
+    if (organisationId) {
+      this.set("organisation_id", organisationId);
+      this.store.peekRecord("gc_organisation", organisationId) ||
+        (await this.store.findRecord("gc_organisation", organisationId, {
+          reload: true
+        }));
+    }
+  },
+
+  setupController(controller, model) {
+    this._super(controller, model);
+    controller.set("user_id", this.get("user_id"));
+    this.initializeOrganisation(controller, model);
+    this.initializeStatus(controller, model);
+  },
+
+  initializeOrganisation(controller, model) {
+    // TODO: Remove gc_organisation model and use only organisation model
+
+    // Currently, the organisation search uses gc_organisation model to find records.
+    // If this model is used, then model.save will pass gc_organisation_id instead of
+    // organisation_id, which is not expected in the API.
+    if (this.get("organisation_id")) {
+      const data =
+        model.get("organisation") ||
+        this.store.peekRecord("gc_organisation", this.get("organisation_id"));
+      controller.set("organisation", data);
+
+      const orgData =
+        this.store.peekRecord("organisation", data.get("id")) ||
+        this.store.createRecord("organisation", {
+          nameEn: data.get("nameEn"),
+          id: data.get("id")
```

Comment:
if there's an `id` shouldn't we `findRecord` instead ??

Reply:
actually, nevermind that, why are we using the `organisations` model at all ? shouldn't we stick to `gc_organisations` ? (which in reality is the real `organisations` on the api side ... 🤦 )